RealmWeaver/RealmAssemblyWeaver.cs

```diff
@@ -0,0 +1,92 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2016 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+using System.Diagnostics;
+using System.Linq;
+using Mono.Cecil;
+using Mono.Cecil.Rocks;
+
+internal static class RealmAssemblyWeaver
+{
+    internal static void PrepareRealmAssemblyForIos(AssemblyDefinition realmAssembly, IAssemblyResolver assemblyResolver)
+    {
+        UpdateWrappersReference(realmAssembly, "__Internal");
+
+        var monotouchModule = assemblyResolver.Resolve("Xamarin.iOS").MainModule;
```

Comment:
Yes it should. All this code is in a very experimental stage so please don't read too much into it just yet. :-)